sodium.cc

```diff
@@ -23,34 +23,34 @@ using namespace v8;
 
 
 // get handle to the global object
-Local<Object> globalObj = NanGetCurrentContext()->Global();
+Local<Object> globalObj = Nan::GetCurrentContext()->Global();
+
+// Retrieve the buffer constructor function
+Local<Function> bufferConstructor =
+       Local<Function>::Cast(globalObj->Get(Nan::New<String>("Buffer").ToLocalChecked()));
 
-// Retrieve the buffer constructor function 
-Local<Function> bufferConstructor = 
-       Local<Function>::Cast(globalObj->Get(NanNew<String>("Buffer")));
-       
 
 // Check if a function argument is a node Buffer. If not throw V8 exception
 #define ARG_IS_BUFFER(i,msg) \
-    if (!Buffer::HasInstance(args[i])) { \
+    if (!Buffer::HasInstance(info[i])) { \
```

Comment:
In nan 2.0, `NAN_METHOD` expands to a method signature with an argument list called `info` (as opposed to args in nan 1). See https://github.com/nodejs/nan/blob/f14e04a3492ede8ca2aac8f049a60c595d7bfbf2/nan.h#L1266 for further information.